elyra/pipeline/component_parameter.py

```diff
@@ -298,6 +299,100 @@ def add_to_execution_object(self, runtime_processor: RuntimePipelineProcessor, e
         runtime_processor.add_disable_node_caching(instance=self, execution_object=execution_object, **kwargs)
 
 
+class CustomSharedMemorySize(ElyraProperty):
+    """An ElyraProperty representing shared memory size for a node."""
+
+    applies_to_generic = True  # custom shared mem size applies to generic components
+    applies_to_custom = True  # custom shared mem size applies to custom components
+
+    property_id = KUBERNETES_SHARED_MEM_SIZE
+    property_display_name = "Shared Memory Size"
+    property_description = """Configure a custom shared memory size in
+    gigabytes (10^9 bytes) for the pod that executes a node. A custom
+    value is assigned if the size property value is a number greater than zero."""
+    property_attributes = [
+        PropertyAttribute(
+            attribute_id="size",
+            display_name="Memory Size (GB)",
+            placeholder=0,
+            input_type="number",
+            hidden=False,
+            required=False,
+        ),
+        PropertyAttribute(
+            attribute_id="units",
+            display_name="Units",
+            input_type="string",
+            hidden=True,
+            required=False,
+        ),
```

Comment:
Does this (and the corresponding validation logic) exist solely in case we decide to expose units someday?  Is there a precedent for this kind of "crystal ball" thinking? :smile:

If so, it seems like the `display_name` value of the _size_ (e.g., `display_name="Memory Size (GB)",`) should also "go through the motions" of conveying the (possibly) variable unit types as well.

Reply:
> exist solely in case we decide to expose units someday

Yep!

> Is there a precedent for this kind of "crystal ball" thinking

I'm not sure there is, but the main benefit is that this serves as a blueprint for the "hidden" property, which was introduced in support of this PR.

> If so, it seems like the display_name value of the size (e.g., display_name="Memory Size (GB)",) should also "go through the motions" of conveying the (possibly) variable unit types as well.

Since there are code changes required to make this property visible (here and in the tests, which for now treat other 'valid unit specifications'  as invalid [1]), I figured the same code change would update the display name by removing the units that are now selected by default and the documentation.